### PR TITLE
Eflow 1.4 LTS and stabilize end to end tests

### PR DESCRIPTION
**Pull request by marcschier**
* Update msi to install 1.4 LTS for eflow
* Provide settings and secrets via .env instead of launchsettings.json to support multiple end to end tests and decouple from UI
* Refactor orchestrated tests to break individual steps into separate test facts
* Read secrets in pipeline correctly
* Better logging and assert messages
* Revert previous changes adding waits as it destabilizes standalone tests (test event processor consistently reads duplicate messages despite previous tests passing)

### CODE REVIEW

**Review by mregen — DISMISSED** [review]

**Review by cristipogacean — DISMISSED** [review]

File: e2e-tests/IIoTPlatform-E2E-Tests/RegistryHelper.cs [hunk 1]
```diff
@@ -52,25 +53,19 @@ public class RegistryHelper : IDisposable {
...
                         return;
                     }
 
-                    // We've observed situations when even after the above waits the module did not yet restart.
-                    // That leads to situations where the publishing of nodes happens just before the restart to apply
-                    // new container creation options. After restart persisted nodes are picked up, but on the telemetry side
-                    // the restart causes dropped messages to be detected. That happens because just before the restart OPC Publisher
-                    // manages to send some telemetry. This wait makes sure that we do not run the test while restart is happening.
-                    await Task.Delay(TestConstants.AwaitInitInMilliseconds, ct).ConfigureAwait(false);
+                    await Task.Delay(TestConstants.DefaultDelayMilliseconds, ct).ConfigureAwait(false);
```
Comment (marcschier) [inline e2e-tests/IIoTPlatform-E2E-Tests/RegistryHelper.cs:60]: Note: reverted previous change which had negative impact on average.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs [hunk 2]
```diff
@@ -75,6 +75,13 @@ IIoTMultipleNodesTestContext context
                 new string[] { ioTHubPublisherDeployment.ModuleName }
             ).ConfigureAwait(false);
 
+            // We've observed situations when even after the above waits the module did not yet restart.
```
Comment (marcschier) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs:78]: Reverted previous changes.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs [hunk 3]
```diff
@@ -83,6 +83,13 @@ IIoTMultipleNodesTestContext context
                 new string[] { ioTHubPublisherDeployment.ModuleName }
             ).ConfigureAwait(false);
 
+            // We've observed situations when even after the above waits the module did not yet restart.
+            // That leads to situations where the publishing of nodes happens just before the restart to apply
+            // new container creation options. After restart persisted nodes are picked up, but on the telemetry side
+            // the restart causes dropped messages to be detected. That happens because just before the restart OPC Publisher
+            // manages to send some telemetry. This wait makes sure that we do not run the test while restart is happening.
+            await Task.Delay(TestConstants.AwaitInitInMilliseconds, _cts.Token).ConfigureAwait(false);
+
```
Comment (marcschier) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/B_PublishMultipleNodesStandaloneDirectMethodTestTheory.cs:92]: Reverted previous changes.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs [hunk 4]
```diff
@@ -82,6 +82,13 @@ IIoTMultipleNodesTestContext context
                 new string[] { ioTHubPublisherDeployment.ModuleName }
             ).ConfigureAwait(false);
 
+            // We've observed situations when even after the above waits the module did not yet restart.
+            // That leads to situations where the publishing of nodes happens just before the restart to apply
+            // new container creation options. After restart persisted nodes are picked up, but on the telemetry side
+            // the restart causes dropped messages to be detected. That happens because just before the restart OPC Publisher
+            // manages to send some telemetry. This wait makes sure that we do not run the test while restart is happening.
+            await Task.Delay(TestConstants.AwaitInitInMilliseconds, cts.Token).ConfigureAwait(false);
```
Comment (marcschier) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs:90]: Reverted previous changes.

**Review by mregen — DISMISSED** [review]

**Comment by mregen** [comment]
just noticed this a draft

**Review by cristipogacean — APPROVED** [review]
